### PR TITLE
🐛 Source Tiktok: unnest `cursor` and `primary key` to the root-level

### LINKED ISSUE #15984: Source Tiktok: un-nest cursor fields
## Current Behavior
As indicated in https://github.com/airbytehq/airbyte/issues/15983 and https://github.com/airbytehq/oncall/issues/406 normalization doesn't support nested cursor fields at the moment and tiktok is failing the append dedupe mode for that reason. This means I can't use the append-dedupe sync mode with tiktok. 

## Expected Behavior
I should be able to use the append-dedupe sync mode with tiktok. The fastest way to achieve this right now is to un-nest the cursor field in tiktok i.e: just move or copy-paste the value of that field to the top-level.

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving:
https://github.com/airbytehq/airbyte/issues/15984
https://github.com/airbytehq/oncall/issues/406

## How
- added `unnest_field` static method to unnest nested fields at runtime
- added new properties to the schemas to reflect changes
- fixed unit_tests up to the latest changes
- tested normalization flow on Postgres

## 🚨 User Impact 🚨
The existing users should `update the schema` to pick up the latest changes.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [X] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [X] Code reviews completed
- [X] Documentation updated
    - [X] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [X] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-tiktok-marketing/unit_tests/unit_test.py [hunk 1]
```diff
@@ -133,8 +133,8 @@ def test_random_items(prepared_prod_args):
 @pytest.mark.parametrize(
     "config, stream_len",
     [
-        (PROD_CONFIG_FILE, 26),
-        (SANDBOX_CONFIG_FILE, 19),
+        (PROD_CONFIG_FILE, 10),
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-tiktok-marketing/unit_tests/unit_test.py]: why did the number of streams change? :thinking:

File: airbyte-integrations/connectors/source-tiktok-marketing/acceptance-test-config.yml [hunk 2]
```diff
@@ -31,11 +26,23 @@ tests:
 
   discovery:
     - config_path: "secrets/prod_config.json"
+      backward_compatibility_tests_config:
+        disable_for_version: "0.1.14"
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-tiktok-marketing/acceptance-test-config.yml:30]: you're not changing the spec here, are you? if I'm not mistaken this config is used for spec backward compatibility tests

File: airbyte-integrations/connectors/source-tiktok-marketing/integration_tests/abnormal_state.json [hunk 3]
```diff
@@ -10,42 +10,50 @@
   },
 
   "ads_reports_daily": {
+    "stat_time_day": "2030-01-01",
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-tiktok-marketing/integration_tests/abnormal_state.json:13]: I believe we should handle this migration gracefully in the code in order not to brake incremental syncs

File: airbyte-integrations/connectors/source-tiktok-marketing/acceptance-test-config.yml (hunk 2 above)
Comment (davydov-d) [reply airbyte-integrations/connectors/source-tiktok-marketing/acceptance-test-config.yml:30]: oh that's discovery, nevermind

**Review by davydov-d — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-tiktok-marketing/unit_tests/unit_test.py (hunk 1 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-tiktok-marketing/unit_tests/unit_test.py]: Local tests produced different numbers, so this was changed. It's already reverted back.

File: airbyte-integrations/connectors/source-tiktok-marketing/integration_tests/abnormal_state.json (hunk 3 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-tiktok-marketing/integration_tests/abnormal_state.json:13]: Updated PR

**Review by davydov-d — APPROVED** [review]

**Review by lazebnyi — APPROVED** [review]

**Pull request merged**